src/mem.rs

```diff
@@ -337,6 +338,16 @@ impl Compress {
         self.inner.compress(input, output, flush)
     }
 
+    /// Similar to [`Self::compress`] but accepts uninitialized buffer.
```

Comment:
If we are adding a new public method, I think it should also be made clear why we'd use one over the other.
Alternatively, we can make it private.

Reply:
Thanks, updated the doc to explain why they should use that